lib/Infer/AliveDriver.cpp

```diff
@@ -156,6 +157,60 @@ struct ReturnLHSRAII {
 };
 }
 
+std::map<souper::Inst *, llvm::APInt>
+performCegisFirstQuery(tools::Transform &t,
+                  std::map<std::string, souper::Inst *> &SouperConsts,
+                  smt::expr &TriedExpr) {
+  IR::Value::reset_gbl_id();
+  IR::State SrcState(t.src);
+  IR::State TgtState(t.tgt);
+  util::sym_exec(SrcState);
+  util::sym_exec(TgtState);
+
+  auto &Sv = SrcState.returnVal();
+  auto &Tv = TgtState.returnVal();
+
+  std::map<souper::Inst *, llvm::APInt> SynthesisResult;
+  SynthesisResult.clear();
+
+  std::set<smt::expr> Vars;
+  std::map<std::string, smt::expr> SMTConsts;
+  for (auto &[Var, Val] : TgtState.getValues()) {
+    auto &Name = Var->getName();
+    if (startsWith("%reserved", Name)) {
+      auto App = Val.first.value.isApp();
+      assert(app);
+      SMTConsts[Name] = (Z3_get_app_arg(smt::ctx(), App, 1));
+    }
+  }
+
+  smt::Solver::check({{(Sv.first.value == Tv.first.value) && (TriedExpr),
```

Comment:
I don't think this is right
@manasij7479 can you look?
this looks like an equivalence check, but we must check refinement

Reply:
Yeah, I have some weird bugs with this check.
Also, this query needs to be explicit about PC (assumes do not seem to be automatically implicated)
@zhengyangl  Is there a way we can avoid this check and use alive's verification as a black box?

Reply:
Yes this is the equivalence check. Need further work on this.

Also another thought is the input specialization should not be done in the Souper land. The query LHS(x_0)=RHS(x_0, C) /\ LHS(x_1)=RHS(x_1, C) /\ LHS(x)=RHS(x, C) is wrong since any poison triggered by specialized input will poison the whole query. We need to handle UB in these specialized queries separately

Reply:
OK-- please add a comment explaining this to this PR
we can figure out how to fix this later

Reply:
@manasij7479 could something like this explain the unsound results you've been seeing/?

Reply:
I don't know.